### ISSUE TITLE
query: re @property-read  & public properties

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#1689]
Am I mis-using `@property-read` in this example?

* https://psalm.dev/r/93ae6c2d68
* using 0525991

```php
<?php
  
/**
* @property-read int $id
*/
class Foo {
  /**
  * @var int
  */
  public $id;
  
  /**
  * @var string
  */
  public $name;
  
  public function __construct(int $id, string $name) {
    $this->id = $id;
    $this->name = $name;
  }
}

$a = new Foo(1, 'foo');

$a->id = 2;
```

i.e. `$a->id = 2;` should cause an error, surely?

**Comment by SignpostMarv** [comment]
see also an example where `@property-read` is ignored: https://psalm.dev/r/b7e84f1e64

**Comment by muglug** [comment]
Yeah, `@property` annotations only affect properties that aren't visible to the caller (i.e. where a `__get`/`__set` would be invoked.

**Comment by SignpostMarv** [comment]
the phpdoc manual says `@property-*` can be used on a class or *interface*, so if we're getting support for key-of/value-of in #762 for magic getters & setters, and a property is defined on an interface but not defined on a class then I was thinking that should also raise an error: https://psalm.dev/r/1a17e68eca

If `@property-read` & `@property-write` are designed to be used with magic getters & settters, should psalm not be notifying the user that the public variable is writable & it should be protected/private?

tl:dr; I'm thinking forward a lot to 7.4 typed properties & how to reduce the overall cruft in code.
